src/port/portindex.tcl

```diff
@@ -45,15 +47,19 @@ proc _read_index {idx} {
     return [list $name $len $line]
 }
 
-proc _write_index {name len line} {
-    global fd
+proc _write_index {name len line write_to_changedports_file} {
+    global fd index_changes changesfd
+    if {$index_changes == 1 && $write_to_changedports_file == 1 } {
```

Comment:
I initially spent some time figuring out if variable `$write_to_changedports_file` is redundant, turns out it's not. :)

Reply:
Yes, it tells the `proc _write_index` to write to the `differences` file.

I have renamed it to `write_diff`.